onnx_tf/handlers/backend/gather_and_scatter_mixin.py

```diff
@@ -0,0 +1,45 @@
+import tensorflow as tf
+
+from onnx_tf.common.tf_helper import tf_shape
+
+
+class GatherAndScatterMixin(object):
+
+  @classmethod
+  def chk_idx_out_of_bounds(cls, data, indices):
+    data_shape = tf_shape(data)
+    indices_shape = tf_shape(indices)
+
+    def _chk_idx_out_of_bounds(i, result):
+      indices_i = tf.transpose(indices)[i]
+      limit_i = tf.cast(data_shape, indices.dtype)[i]
+      cond1 = tf.greater_equal(indices_i, tf.negative(limit_i))
+      cond2 = tf.less(indices_i, limit_i)
+      result = tf.reduce_all(tf.logical_and(cond1, cond2))
+      return i + 1, result
+
+    _, result = tf.while_loop(
+        lambda i, result: tf.logical_and(tf.less(i, indices_shape[-1]), result),
+        _chk_idx_out_of_bounds, [0, True])
+    return result
```

Comment:
Please provide documentation for all class methods to make the functionality very clear. Check ConvMixin as a reference. Because these mixin classes are to be used by multiple handlers, we want to make them as easy to understand and reuse as possible.

Reply:
@chinhuang007 please review again, thanks.